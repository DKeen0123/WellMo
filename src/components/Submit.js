import React from 'react';
import Button from './styledComponents/SubmitButton';

const Submit = props => {
  return (
    <Button id="displayButton" onClick={props.handleDisplayClick}>
      Enter Mood
    </Button>
  );
};

export default Submit;
