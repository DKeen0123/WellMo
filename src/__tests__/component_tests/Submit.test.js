import React from 'react';
import { shallow, mount } from 'enzyme';
import Submit from '../../components/Submit';

describe('Submit', () => {
  let submit = shallow(<Submit />);

  it('renders correctly', () => {
    expect(submit).toMatchSnapshot();
  });

  describe('when a user clicks the button', () => {
    it('fires the `handleDisplayClick()` function', () => {
      const mockhandleDisplayClick = jest.fn();
      submit = shallow(<Submit handleDisplayClick={mockhandleDisplayClick} />);
      submit.find('#displayButton').prop('onClick')();
      expect(mockhandleDisplayClick).toBeCalled();
    });
  });
});
