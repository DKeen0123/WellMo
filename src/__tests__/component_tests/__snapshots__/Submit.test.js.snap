// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Submit />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Enter Mood",
      "id": "displayButton",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "Enter Mood",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Enter Mood",
        "id": "displayButton",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Enter Mood",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
